pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -422,11 +429,13 @@ def get_layers(self, distance_threshold=0.01, id_list=None, wrap_atoms=True, pla
             positions = np.einsum('ij,i,nj->ni', mat, 1/np.linalg.norm(mat, axis=-1), positions)
         layers = []
         for ii, x in enumerate(positions.T):
-            cluster = AgglomerativeClustering(
-                linkage='complete',
-                n_clusters=None,
-                distance_threshold=distance_threshold
-            ).fit(x.reshape(-1, 1))
+            if cluster_method is None:
```

Comment:
Does `cluster_method` have to be defined within the `for`-loop?

Reply:
No, should be fine outside, too.  But in any case creating a new `AgglomerativeClustering`, etc. should be cheap. I'll just move it though, for symmetry.